src/components/views/messages/DateSeparator.tsx

```diff
@@ -34,13 +51,40 @@ function getDaysArray(): string[] {
 }
 
 interface IProps {
+    roomId: string,
     ts: number;
     forExport?: boolean;
 }
 
+interface IState {
+    contextMenuPosition?: DOMRect
+}
+
 @replaceableComponent("views.messages.DateSeparator")
```

Comment:
I think I've nailed the keyboard interaction with the Jump to date headers now:

 1. `ContextMenu` is now using `RovingTabIndex` so no focus-holes and standard with the rest of Element. Navigable by `Arrow` keys
 1. Pressing `Tab` will close the `ContextMenu`  (maintains previous page focus)
 1. If you're on the "Jump to date" menu option, you can `Tab` and `Shift + Tab` along the form and controls.
 1. If you press `Tab` off the end of the form, it will close the `ContextMenu` (maintains previous page focus)

https://user-images.githubusercontent.com/558581/145678920-ee9a2393-f95f-46dc-82c7-aff761efccc9.mov

Reply:
This is now updated to use the standard `RovingTabIndex` `ContextMenu` behavior established in https://github.com/matrix-org/matrix-react-sdk/pull/7353

 - Arrow keys to navigate items up and down
 -  `Tab`/`Shift + Tab` will normally close the `ContextMenu` and move onto the next focusable item on the page
 - But inside a `ContextMenu` , once you navigate to an `<input>`, arrow keys act on the `<input>` text instead and you can `Tab`/`Shift + Tab` out of it to the next `ContextMenu` item